### DISCUSSION TITLE
Turret lock feature for turreted voxels units

### DISCUSSION THREAD

**Discussion by Crazybird2** [Phobos-developers/Phobos#1036] [Ideas and feedback]
### First check

- [X] The suggestion is **NOT** related to functionality of Ares, HAres, Kratos or any other YR engine extension.
- [X] The suggested feature / enhancement is **NOT** implemented in Ares/Phobos yet and the existing possibilites are not viable or good enough to achieve the wanted result.
- [X] I agree to elaborate the details if requested and provide thorough testing if the feature is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I thought out how the feature / enhancement should work and described it in the most detailed and clear way.

### Description

turreted voxels vehicles can benefit from turret animation to create charging or recoil effects. I want to request a feature that allows you to lock the turret of of a voxels units in place but still be able to use the IsChargeTurret=yes and IsGattling=yes turret animation logic.

### INI code

```ini
[SomeUnit]
Turret.Lock=yes
or
TurretRotation.Lock=yes
```

**Comment by Daiki9527** [comment]
What a coincidence,that's exactly the function what i want to pull.

Besides,Units with IsChargeTurret only fire Weapon1 and EliteWeapon1,which is a limitation or a shortcoming.
Or,we can expand this function like:

IsChargeTurret=yes
WeaponCount=2
TurretCount=7
Weapon1=xx1
EliteWeapon1=xx1E
Weapon2=xx2
EliteWeapon1=xx2E

;different charge turrets for different weapon
ChargeTurret.DisplayWeapon1Index=1,2,3
ChargeTurret.DisplayWeapon2Index=4,5,6

;Formula be like:
WeaponCount=X
TurretCount=X+1;plus the original turret
WeaponX=
EliteWeaponX=
ChargeTurret.DisplayWeaponXIndex=
